sympy/polys/polytools.py

```diff
@@ -7938,3 +7938,134 @@ def named_poly(n, f, K, name, x, polys):
     else:
         poly = Poly.new(poly, head)
     return poly if polys else poly.as_expr()
+
+
+def factor_system_poly(polys, **kwargs):
+
+    if not polys:
+        return [[]]
+
+    factored_eqs = []
+    for poly in polys:
+        if not isinstance(poly, Poly):
+            raise TypeError("All inputs must be Poly instances")
+
+        try:
+            constant, factors = factor_list(poly, **kwargs)
+            if not factors and constant.is_zero:
+                return []
+            elif not factors:
+                continue
```

Comment:
This is not correct:
```python
In [2]: factor_system([0, x])  # should be [{x}]
Out[2]: []

In [3]: factor_system([1, x])  # should be something like [{1}]
Out[3]: [{x}]
```

Reply:
Please add any cases like this that are discussed to the tests in the PR.

Reply:
I don't understand why expected output of `factor_system([1, x])` is `[{1}]`

Reply:
Okay maybe `factor_system([1, x])` should be `[]`.

The definition of what `factor_system` returns is that if `E1, E2, ... = factor_system(E)` then the solutions of equations `E` are the union of the solutions of `E1`, `E2`, ... If we have an equation `1` (i.e. `1 = 0`) then there are no solutions to equations `E` and so returning an empty list means that the solutions are an empty union:
```python
In [2]: Union(*[])
Out[2]: ∅
```
An output of `[{1}]` was supposed to indicate that there are no solutions since the only set of solutions is the solutions of `{1}` which is the empty set. I see now that it is better just to return `[]` though.

If the whole return value is `[]` then that means that there are no solutions. On the other hand an empty subsystem like `[[]]` would mean that anything is a solution since any value of the variables can satisfy the empty equation list. Hence:
```python
factor_system([0])  ->  factor_system([])  ->  [[]]
factor_system([0, x])  ->  factor_system([x])  ->  [{x}]
factor_system([1, x])  ->  factor_system([1])  ->  []
```

Reply:
In conclusion we have,

i) If `gens` only constitutes `0` then `factor_system` should return `[{}]`  showing that the system has infinite solutions. 

ii) If "gens' constitutes `0` along with another subsystem then it is equivalent to calling `factor_system` with that subsystem alone.

iii) If 'gens constitutes any non-zero constant then `factor_system` should return `[ ]` which shows the system has no solutions.